src/api/rest/v1/db/controller.ts

```diff
@@ -40,6 +41,98 @@ export class DbController {
         }
     }
 
+    public async create(req: Request, res: Response) {
+        try {
+            const { context } = await Utils.getNetworkConnectionFromRequest(req)
+            const dbName = req.params.database
+
+            const permissions = Utils.buildPermissions(req)
+            const db = await context.openDatabase(dbName, {
+                // @ts-ignore
+                permissions
+            })
+
+            const record = req.body.record
+            const options = req.body.options || {}
+            const result = await db.save(record, options)
+
+            if (result) {
+                const savedRecord = await db.get((<any> result).id, {})
+                record._rev = (<any> result).rev
+                res.json({
+                    success: true,
+                    record: savedRecord
+                })
+            } else {
+                res.json({
+                    success: false,
+                    // @ts-ignore
+                    errors: db.errors
+                })
+            }
+        } catch (error) {
+            const message = error.message
+
+            res.status(500).send({
+                error: message
+            });
+        }
+    }
+
+    public async update(req: Request, res: Response) {
+        try {
+            const { context } = await Utils.getNetworkConnectionFromRequest(req)
+            const dbName = req.params.database
+
+            const permissions = Utils.buildPermissions(req)
+            const db = await context.openDatabase(dbName, {
+                // @ts-ignore
+                permissions
+            })
+            const rowId = req.params.recordId
+
+            const record = req.body.record
+            record._id = rowId
+            const options = req.body.options || {}
+            const result = await db.save(record, options)
+
+            // Ensure the record exists
+            try {
+                const existingRecord = await (db.get(rowId, {}))
+            } catch (err: any) {
+                // Record doesn't exist
+                return res.status(404).json({
+                    success: false,
+                    message: "Not found"
+                })
+            }
+
+            if (result) {
+                const savedRecord = await ds.get(record._id, {})
```

Comment:
Should be `db`, copy/paste error, CI failing

Reply:
Doh